File: yu-ai-router-sdk/src/test/java/com/yupi/ai/sdk/example/SimpleExample.java
/**
 * 简单示例
 *
 * @author <a href="https://codefather.cn">编程导航学习圈</a>
 */
package com.yupi.ai.sdk.example;

import com.yupi.ai.sdk.YuAIClient;
import com.yupi.ai.sdk.model.ChatResponse;

/**
 * 最简单的使用示例
 */
public class SimpleExample {

    public static void main(String[] args) {
        // 创建客户端
        YuAIClient client = YuAIClient.builder()
                .apiKey("sk-d8483dd850cf44729f41cb59f7bc1e1d")  // 替换为你的 API Key
                .baseUrl("http://localhost:8123/api")
                .build();

        try {
            // 同步调用
            ChatResponse response = client.chat("你好，请介绍一下自己");
            System.out.println("响应: " + response.getContent());

            // Token 使用统计
            System.out.println("\nToken 统计:");
            System.out.println("输入: " + response.getUsage().getPromptTokens());
            System.out.println("输出: " + response.getUsage().getCompletionTokens());
            System.out.println("总计: " + response.getUsage().getTotalTokens());

        } finally {
            // 关闭客户端
            client.close();
        }
    }
}

File: yu-ai-router-python-sdk/examples/simple_example.py
"""Simple SDK usage example."""

from __future__ import annotations

import os

from yu_ai_router_sdk import YuAIClient


def main() -> None:
    api_key = os.getenv("YU_AI_API_KEY", "")
    base_url = os.getenv("YU_AI_BASE_URL", "http://localhost:8123/api")
    client = YuAIClient.builder().api_key(api_key).base_url(base_url).build()
    try:
        response = client.chat("你好，请介绍一下自己")
        print(f"响应: {response.content}")
        print("\nToken 统计:")
        print(f"输入: {response.usage.prompt_tokens}")
        print(f"输出: {response.usage.completion_tokens}")
        print(f"总计: {response.usage.total_tokens}")
    finally:
        client.close()


if __name__ == "__main__":
    main()
